boards/PDM/Inc/Io/Io_Efuse_22XS4200.h

```diff
@@ -0,0 +1,646 @@
+    StatusType_OV      = (1 << 8)  //!< Over Voltage Fault
+} StatusType_e;
+
+//! Enumeration for the fault types
+typedef enum
+{
+    FaultType_NoFault = (0 << 0), //!< No fault
+    FaultType_OC      = (1 << 0), //!< Over Current fault
+    FaultType_SC      = (1 << 1), //!< Severe Short Circuit fault
+    FaultType_OT      = (1 << 2), //!< Over Temperature fault
+    FaultType_OS      = (1 << 3), //!< Output Shorted to Vpwr fault
+    FaultType_OLOFF   = (1 << 4), //!< Open Load in OFF State fault
+    FaultType_OLON    = (1 << 5), //!< Open Load in ON State fault
+    FaultType_OTW     = (1 << 8)  //!< Over Temperature Warning fault
+} FaultType_e;
+
+//! Enumeration for the PWM types
+typedef enum
+{
+    PWM_Type_DC_0 = (1 << 0), //!< PWM Duty Cycle bit 0
+    PWM_Type_DC_1 = (1 << 1), //!< PWM Duty Cycle bit 1
+    PWM_Type_DC_2 = (1 << 2), //!< PWM Duty Cycle bit 2
+    PWM_Type_DC_3 = (1 << 3), //!< PWM Duty Cycle bit 3
+    PWM_Type_DC_4 = (1 << 4), //!< PWM Duty Cycle bit 4
+    PWM_Type_DC_5 = (1 << 5), //!< PWM Duty Cycle bit 5
+    PWM_Type_DC_6 = (1 << 6), //!< PWM Duty Cycle bit 6
+    PWM_Type_DC_8 = (1 << 7), //!< PWM Duty Cycle bit 7
+    PWM_Type_ON   = (1 << 8)  //!< PWM ON
+} PWM_Type_e;
+
+//! Enumeration for the channel configurations types
+typedef enum
+{
+    ChannelConfigType_Delay_0              = (1 << 0), //!< Switch-on Delay bit 0
+    ChannelConfigType_Delay_1              = (1 << 1), //!< Switch-on Delay bit 1
+    ChannelConfigType_Delay_2              = (1 << 2), //!< Switch-on Delay bit 2
+    ChannelConfigType_SlewRate_0           = (1 << 3), //!< Slew Rate bit 3
+    ChannelConfigType_SlewRate_1           = (1 << 4), //!< Slew Rate bit 4
+    ChannelConfigType_DirecControl_Disable = (1 << 5), //!< Direct Control Disable
+    ChannelConfigType_OpenLoad_InOffState  = (1 << 6), //!< Open Load Detection in Off-state
+    ChannelConfigType_OpenLoad_InOnState   = (1 << 7), //!< Open Load Detection in On-state
+    ChannelConfigType_ShortCircuitDetection_Disable = (1 << 8) //!< Short-circuit to Vpwr Detection Disable
+} ChannelConfigType_e;
+
+struct E_Fuse
+{
+    bool         watch_dog_state;
+    StatusType_e status;
+    struct ChannelFaults
+    {
+        FaultType_e channel_0_faults;
+        FaultType_e channel_1_faults;
+    } faults;
+    struct E_FuseChipSelect
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } chip_select;
+    struct E_FuseCurrentSenseSync
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } current_sense_sync;
+};
+
+/* Function Definitions */
+
+/**
+ * Initializes the SPI handle for the Efuses
+ * @param hspi Handle to the SPI peripheral used for the Efuses
+ */
+void Io_Efuse_22XS4200_Init(SPI_HandleTypeDef *const hspi);
+
+/**
+ * Configures the Efuse's registers using the values defined in: REGNAME_CONFIG
+ * @param *e_fuse Pointer to E_Fuse structure being configured
```

Comment:
Applies for other occurrences obviously
```suggestion
 * @param e_fuse Pointer to E_Fuse structure being configured
```